meetings/2022-09-28.md

```diff
@@ -0,0 +1,187 @@
+* Elf: https://github.com/solid/solid-oidc/blob/main/alternative-flows.md
+* Elf: What do you use for authentication there?
+* Matthieu: That's the thing. I don't really want to discuss Solid OIDC, but I want to acknowledge the fact that an Authorization Server might receive a request in a client-less context. And specific rules should be possible to express against that kind of scenarios (in other words, being able to say in ACP that it is different to have 1 or no client).
+* Matthieu: An "AuthenticatedClient" matcher will fail if there is not a client. A "PublicClient" matcher will succeed if there is not a client.
+* Elf: So it's like "AuthenticatedAgent", the agent needs to be present if "PublicAgent" it will not need to have an agent.
+* ...: Ok, so ACP uses public agent in the same sort of way as WAC?
+* Matthieu: Yes.
+* Elf: This sounds like [my comment](https://github.com/solid/authorization-panel/issues/292#issuecomment-1063600607) on secure defaults for client access issue.
+* Elf: What happens when a matcher does not specify a client attribute compared to the case where the matcher specifies a client attribute matching the "PublicClient"?
+* Matthieu: If a matcher specifies a single client attribute matching the "PublicClient", then it will be satisfied if any or no client is used (in other words, all contexts).
+* ...: If a matcher does not specify a client attribute, then it will need to match on something else (for example, agent or issuer...). In other words, a Matcher that does not specify a client attribute will not be satisfied by a client but might be satisfied by something else.
+* Matthieu: Via Authenticated client now you could say that a client is required (disallow unmitigated access). Consider the following policy:
+
+```ttl
+[]
+  acp:deny acl:Read, acl:Append, acl:Write, acl:Control ;
+  acp:allOf [ acp:client acp:PublicClient ] ;
+  acp:noneOf [ acp:client acp:AuthenticatedClient ] .
+```
+
+* Matthieu: In that example policy, all access is denied except when there is a client. That addresses [your comment](https://github.com/solid/authorization-panel/issues/292#issuecomment-1063600607) in some way because it is a way of denying unmitigated client access.
+* Elf: But isn't the deny a default?
+* Matthieu: The default is that nothing is allowed.
+* ...: Not that everything is denied.
+* ...: Deny acts as a thing that trumps everything that is allowed (something denied cannot be otherwise allowed).
+* Elf: Let's look for example at the [client constraint evaluation](https://github.com/solid/authorization-panel/blob/main/proposals/evaluation/uc-2.5.2-client-constraints.md) (simplified): 
+
+```ttl
+PREFIX acp: <http://www.w3.org/ns/solid/acp#>
+PREFIX acl: <http://www.w3.org/ns/auth/acl#>
+PREFIX ex: <https://example.com/>
+
+ex:ReaderAgentPolicy
+  acp:anyOf ex:BobMatcher ;
+  acp:allow acl:Read .
+
+ex:BobMatcher
+  acp:agent ex:Bob ;
+  acp:client ex:Projectron .
+
+```
```

Comment:
As above, every line from 107 to 120 needs to be indented with 2 (additional) space chars, so that block gets rendered under/within Elf's `(simplified):` bullet. I cannot submit this as a suggestion, because GitHub can't handle the 3-backtick code fences within a suggestion.

(Secondarily, I would also delete line 119, just for cleanliness.)